api/common_test.go

```diff
@@ -0,0 +1,175 @@
+package api
+
+import (
+	"io/ioutil"
+	"path/filepath"
+	"testing"
+
+	"github.com/docker/docker/api/types"
+)
+
+type ports []types.Port
+
+// DisplayablePorts
+func TestDisplayablePorts(t *testing.T) {
+	cases := map[*ports]string{
```

Comment:
Yes, that's what I had in mind.

I'm really picking nits here, but the indentation looks a little strange. Some lines which should be at the same level of indentation are at different levels. Maybe if

    {[]types.Port{{

became

    {
        []types.Port{
            {

...it would be easier to read.